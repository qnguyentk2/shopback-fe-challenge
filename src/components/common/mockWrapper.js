import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import branches from '../../reducers/branches';

const mockStore = configureStore([thunk]);

export default function getWrapper(
  component,
  useMemory = false,
  useShallow = false,
) {
  const store = mockStore({
    branches,
  });
  if (useShallow) {
    return shallow(<Provider store={store}>{component}</Provider>);
  }
  if (!useMemory) {
    return mount(<Provider store={store}>{component}</Provider>);
  }
  return mount(
    <Provider store={store}>
      <MemoryRouter>{component}</MemoryRouter>
    </Provider>,
  );
}
