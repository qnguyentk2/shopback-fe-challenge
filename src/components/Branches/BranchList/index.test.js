import React from 'react';
import mockWrapper from '../../common/mockWrapper';
// import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import '../../../setupTests';

import Index from './index';

test('renders without crashing', () => {
  const wrapper = mockWrapper(<Index />, true);
  expect(toJSON(wrapper)).toMatchSnapshot();
  // expect(wrapper.html()).toMatch(/Popular/);
});