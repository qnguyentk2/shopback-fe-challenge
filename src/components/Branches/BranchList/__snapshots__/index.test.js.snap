// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "8zao6y",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "8zao6y",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <BranchList
        PropTypes={false}
        categoryId={0}
        categoryName=""
        stores={Array []}
      >
        <div
          className="branch-list"
        >
          <div
            className="branch-list__button"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              View All  stores
            </button>
          </div>
        </div>
      </BranchList>
    </Router>
  </MemoryRouter>
</Provider>
`;
